docs/application/native/guides/applications/component-based-app.md

```diff
@@ -0,0 +1,618 @@
+| `service_component_restore_content_cb()`    | Used to restore instance's latest state. The data stored in `service_component_save_content_cb` will be passed by parameter. |
+| `service_component_start_command_cb()`    | Used to start an instance. Requested `app_control_h` will be passed by parameter. |
+| `service_component_save_content_cb()`    | Used to take necessary actions when there is a need to store or restore data for launching the next instance. This callback will be called right before `service_component_destroy_cb`|
+| `service_component_destroy_cb()` | Used to take necessary actions when the frame-component instance is terminating. Releases all resources, especially the allocated and shared resources. |
+| `service_component_action_cb()` | Used to take necessary actions when another application sends a launch request.  To receive action event you must [register action](#register_action)|
+
+The service-component instance can also receive some basic system events. The following table shows available system events callbacks:
+
+| Callback                             | Description                              |
+|----------------------------------------|------------------------------------------|
+| `service_component_low_memory_cb`                 | Event type for the callback function that is responsible for saving data from the main memory to a persistent memory or storage to avoid data loss. Data loss can occur if the Tizen platform low memory killer kills your application to free more memory. The callback function must release any cached data in the main memory to secure more free memory. |
+| `service_component_low_battery_cb`                | Event type for the callback function that is responsible for saving data from the main memory to a persistent memory or storage to avoid data loss in case of complete power failure. The callback function must also stop heavy CPU consumption or power consumption activities to save the remaining power. |
+| `service_component_device_orientation_changed_cb` | Event type for the callback function that is responsible for changing the display orientation to match the device orientation. |
+| `service_component_language_changed_cb`           | Event type for the callback function that is responsible for refreshing the display into a new language. |
+| `service_component_region_format_changed_cb`      | Event type for the callback function that is responsible for refreshing the display into a new time zone. |
+| `service_component_suspended_state_changed_cb`    | Event type for the callback function that is responsible for taking necessary actions before entering the suspended state or after exiting from the state. |
+
+
+
+<a name="attribute"></a>
+## Application Attributes
+
+Describe your component-based application attributes in the manifest file. The attributes determine the application behavior. The following code example illustrates how you can define the attributes:
+
+```
```

Comment:
Global:
Please verify and apply the proper code indentation of four spaces for all the code blocks in the document.

Reply:
Done